homeassistant/util/__init__.py

```diff
@@ -309,3 +309,53 @@ def wrapper(*args, **kwargs):
                 throttle[0].release()
 
         return wrapper
+
+
+class RetryOnError(object):
+    """A class for retrying a failed task a certain amount of tries.
+
+    This method decorator makes a method retrying on errors. If there was an
+    uncaught exception, it schedules another try to execute the task after a
+    retry delay. It does this up to the maximum number of retries.
+
+    It can be used for all probable "self-healing" problems like network
+    outages. The task will be rescheduled using HAs scheduling mechanism.
+
+    It takes a Hass instance, a maximum number of retries and a retry delay
+    in seconds as arguments.
+    """
+
+    def __init__(self, hass, retry_limit=0, retry_delay=20):
```

Comment:
0 is actually "higher" - as this tells how long to wait before the next retry. So 0 would meas "instant retry" - which makes no sense most of the time.